ERCS/erc-6900.md

```diff
@@ -622,11 +622,21 @@ See `https://github.com/erc6900/reference-implementation`
+From the wallet side, there are certain checks that the wallet/SDK should perform on the UI/UX side.
 
-Execution hooks have no awareness of other execution hooks being performed in the same function selector execution setting. Since execution hooks can perform state changes, this reveals an important security consideration: An execution hook can only assure that at the time of its own execution, certain conditions are met, but this can not be generalized to the entire pre execution context of potentially multiple pre execution hooks. For example, a pre execution hook cannot be assured that the storage it performed validation upon does not get further updated in subsequent pre execution hooks. Even a post execution hook potentially repeating the validation cannot assure that the storage remains unmodified because a prior post execution hook may have reset the state. As long as the requirements checked by a module as part of an execution hook are only modifiable by the module itself, this can be considered safe.
+The standard introduces a concept of the execution manifest, which describes the execution functions, interface IDs, and hooks that should be installed on the account from an execution manifest. As part of constructing parameters to `installExecution()`, the SDK creates an `executionManifest` parameter that will specify the actions of the execution module being installed. SDKs will process the provided `executionManifest()` function of the underlying module, but the finished parameter will not necessarily be identical to the module’s return value of `executionManifest()`. Furthermore, this parameter is only going through very limited checks on-chain as part of the installation process. Therefore, the `executionManifest` parameter needs to be carefully constructed and verified before being used for installations. Furthermore, the `executionManifest` parameter of uninstallation should ideally be the same used in installation to not leave residual data.
+
+The standard supports a `isGlobalValidation` flag for validation functions, which means that this function is added to a global validation pool and can validate any of the execution functions that expose themselves to global validation via `allowGlobalValidation` flag. By default all native execution functions are allowed to be globally validated. Therefore, the wallets should be careful on what validation functions they allow to be installed that have global validation allowed, as that would allow these functions to validate all native functions by default and hence bypass any restrictions that have been added to protect these native/execution functions. In a sense these global validation functions gain root access to the account.
```

Comment:
> By default all native execution functions are allowed to be globally validated.

I don't think this is true in the general case - it is up to the implementation to choose which native functions fall into this category. We should update the wording here.

Reply:
That is a good point. I was going off the reference implementation, which has all native functions under global validation, but that doesn't have to be the case.